### PR TITLE
feat: add cache version

### PR DESCRIPTION
**Pull request by PerseidMeteor**
add cache version information

changed:
- add cache version and check if version matched

### CODE REVIEW

File: pkg/cache/cache.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ import (
 	"github.com/pkg/errors"
 )
 
+const LayerAnnotationCacheVersion = "containerd.io/snapshot/nydus-cache-version"
```
Comment (PerseidMeteor) [inline pkg/cache/cache.go:43]: `LayerAnnotationCacheVersion` may need add to constant of nydus snapshotter

File: pkg/converter/converter.go [hunk 2]
```diff
@@ -116,6 +116,8 @@ func (cvt *Converter) Convert(ctx context.Context, source, target, cacheRef stri
 			if err != nil {
 				if errors.Is(err, ctrErrdefs.ErrNotFound) {
 					logger.Infof("cache %s not found", cacheRef)
+				} else if errors.Is(err, ctrErrdefs.ErrNotImplemented) {
+					logger.Warnf("unsupported cache version")
```
Comment (imeoer) [inline pkg/converter/converter.go:120]: logger.Warnf("ignore cache %s, unmatched version: %s, expected: %s")

File: pkg/cache/cache.go [hunk 3]
```diff
@@ -418,6 +427,8 @@ func (rc *RemoteCache) update(ctx context.Context, orgDesc, newDesc, cacheDesc *
 			Platform:  platform,
 		})
 	}
+	// append cache version annotation
+	cacheIndex.Annotations[LayerAnnotationCacheVersion] = rc.version
```
Comment (imeoer) [inline pkg/cache/cache.go:431]: Should add the annotation to manifest instead of index?

File: pkg/cache/cache.go (hunk 1 above)
Comment (imeoer) [reply pkg/cache/cache.go:43]: It's ok to place here now.

File: pkg/converter/converter.go (hunk 2 above)
Comment (PerseidMeteor) [reply pkg/converter/converter.go:120]: We can log in `cache.Fetch`

File: pkg/cache/cache.go (hunk 3 above)
Comment (PerseidMeteor) [reply pkg/cache/cache.go]: ok

File: misc/config/config.nydus.ref.yaml [hunk 4]
```diff
@@ -34,6 +34,8 @@ provider:
       threshold: 1000MB
   # remote cache record capacity of converted layers, default is 200.
   cache_size: 200
+  # remote cache version
```
Comment (imeoer) [inline misc/config/config.nydus.ref.yaml:37]: remote cache version, the remote cache will be discarded if uses a version not matched with this.

Comment (PerseidMeteor) [reply misc/config/config.nydus.ref.yaml]: `remote cache version, cache in remote must match the specified version, or discard cache.`

**Review by imeoer — APPROVED** [review]

**Pull request merged**